src/site/content/en/react/how-to-use-amp-in-nextjs/index.md

```diff
@@ -0,0 +1,208 @@
+---
+layout: codelab
+title: How to use AMP in Next.js
+authors:
+  - houssein
+subhead: |
+  Try out the two ways that you can add AMP to a Next.js app
+date: 2019-10-17
+draft: true
+glitch: next-amp-start
+related_post: how-amp-can-guarantee-fastness-in-your-nextjs-app
+---
+
+## What will you learn?
+
+This codelab lets you try out the two ways that you can use AMP in a Next.js app.
+Check out [How AMP can guarantee fastness in your Next.js app][guidance] to learn why you
+might want to add AMP support to your Next.js app.
+
+### How to create Hybrid AMP pages {: #hybrid }
+
+{% Aside 'caution' %}
+  The [AMP-only approach](#amponly) is the recommended path for using AMP with Next.js. The
+  Hybrid AMP approach described in this section has a higher maintenance cost than the AMP-only
+  approach because it requires you to maintain two versions of each page. You should only use the
+  Hybrid approach if you're certain that the AMP-only approach won't work.
+{% endAside %}
+
+The **Hybrid AMP** approach creates an accompanying AMP version of any Next.js page. The regular
+page can always be accessed by your users but search engines will surface and cache the AMP version
+of the page instead.
+
+There are multiple ways to configure how Next.js renders and serves pages. Using a `config`
+object allows you to modify these on a per-page basis. In order to serve a specific page as
+an AMP page, you need to export the `amp` property in the object:
+
+```jsx/2
+import React from 'react'
+
+export const config = { amp: 'hybrid' };
+
+const Home = () => (
+  <p>This is the home page</p>
+);
+  
+export default Home;
+```
+
+{% Instruction 'remix', 'ol' %}
+
+{% Instruction 'preview', 'ol' %}
+
+1. Add `?amp=1` to the end of the URL. The page looks the same, but if you look in the
+   Console you'll see that the AMP version of the page is being rendered.
+
+<figure>
+  <img src="hybrid.png"
+       alt="The live page and a message in the Chrome DevTools Console stating that the page is 
+            powered by AMP.">
+  <figcaption>
+    The AMP version of the page.
+  </figcaption>
+</figure>
+
+Since the page only has a single `<p>` tag, there's no visible difference between the
+main page and its AMP version. The Hybrid AMP approach is useful when you need to render
+AMP-specific components only when the AMP page is requested.
```

Comment:
(This wording might be obsolete now, but just for reference the Hybrid AMP approach is frequently used when there is some experience on the original site that cannot be supported by AMP (or not easily supported). The canonical HTML page has the full JavaScript experience and the AMP page might have a slightly degraded experience if it cannot be ported to AMP. With the introduction of <amp-script> however, the number of such cases has gone down so the reason for hybrid AMP is also less strong than it used to be.)